soc/nordic/nrf52/CMakeLists.txt

```diff
@@ -3,8 +3,8 @@
 zephyr_library_sources(soc.c)
 zephyr_include_directories(.)
 
-if(CONFIG_SOC_NRF52832_ALLOW_SPIM_DESPITE_PAN_58 AND CONFIG_SPI_NRFX_SPIM)
-  message(WARNING "Both SOC_NRF52832_ALLOW_SPIM_DESPITE_PAN_58 and an NRF SPIM driver are enabled, therefore PAN 58 will apply if RXD.MAXCNT == 1 and TXD.MAXCNT <= 1")
+if(CONFIG_NRF52_ANOMALY_58_WORKAROUND AND CONFIG_SPI_NRFX_SPIM)
+  message(WARNING "Both CONFIG_NRF52_ANOMALY_58_WORKAROUND and an NRF SPIM driver are enabled, therefore PAN 58 will apply if RXD.MAXCNT == 1 and TXD.MAXCNT <= 1")
```

Comment:
This doesn't really make sense. The warning that anomaly 58 can occur should be issued when the workaround is NOT enabled.

Reply:
Previous logic was:
- If `ALLOW_DESPITE_PAN` is not enabled, the driver will not build.
- If it is enabled, it will apply workaround and build but print this warning.

My understanding is that changes in this file preserve behaviour from second bullet point (if `WORKAROUND` is enabled).
What is not preserved is the first bullet point, but I am not sure if that is necessary.

Reply:
> Previous logic was:
> 
> * If `ALLOW_DESPITE_PAN` is not enabled, the driver will not build.
> * If it is enabled, it will apply workaround and build but print this warning.

The workaround was applied only for those instances that had the `anomaly-58-workaround` property set in dts. Hence the warning that for the others the anomaly would occur.

Reply:
How about removing it completely as it is now handled in a standard way and users should be aware of anomalies?

Reply:
Sounds good. The workaround is enabled by default, so users deciding to disable it should know what they're doing.

Reply:
Removed